osquery/tables/yara/yara.cpp

```diff
@@ -97,6 +104,63 @@ static YARAConfigParser getYaraParser(void) {
   return yaraParser;
 }
 
+bool isRuleUrlAllowed(std::set<std::string> signature_set, std::string url) {
+  Uri test_uri(url);
+  for (const auto& sig : signature_set) {
+    Uri sig_uri(sig);
```

Comment:
I put it inside `try/catch` block because `Uri` may throw an exception if the URL provided with the query is invalid.

https://github.com/osquery/osquery/blob/master/osquery/remote/uri.cpp#L43